tests/beaker_tests/cisco_ospf_area_vlink/test_ospf_area_vlink.rb

```diff
@@ -0,0 +1,111 @@
+  operating_system: 'nexus',
+  resource_name:    'cisco_ospf_area_vlink',
+}
+
+# Test hash test cases
+tests[:default] = {
+  desc:           '1.1 Defaults',
+  title_pattern:  'dark_blue default 1.1.1.1 2.2.2.2',
+  preclean:       'cisco_ospf',
+  manifest_props: {
+    auth_key_chain:                     'default',
+    authentication:                     'default',
+    authentication_key_encryption_type: 'default',
+    authentication_key_password:        'default',
+    dead_interval:                      'default',
+    hello_interval:                     'default',
+    message_digest_algorithm_type:      'default',
+    message_digest_encryption_type:     'default',
+    message_digest_key_id:              'default',
+    message_digest_password:            'default',
+    retransmit_interval:                'default',
+    transmit_delay:                     'default',
+  },
+  code:           [0, 2],
+  resource:       {
+    authentication_key_encryption_type: 'cleartext',
+    dead_interval:                      40,
+    hello_interval:                     10,
+    message_digest_algorithm_type:      'md5',
+    message_digest_encryption_type:     'cleartext',
+    message_digest_key_id:              0,
+    retransmit_interval:                5,
+    transmit_delay:                     1,
+  },
+}
+
+# Non-default Tests. NOTE: [:resource] = [:manifest_props] for all non-default
+
+tests[:non_default] = {
+  desc:           '2.1 Non Defaults',
+  title_pattern:  'dark_blue default 1.1.1.1 2.2.2.2',
+  preclean:       'cisco_ospf',
+  manifest_props: {
+    auth_key_chain:                     'testKeyChain',
+    authentication:                     'md5',
+    authentication_key_encryption_type: '3des',
+    authentication_key_password:        '3109a60f51374a0d',
+    dead_interval:                      '500',
+    hello_interval:                     '2000',
+    message_digest_algorithm_type:      'md5',
+    message_digest_encryption_type:     'cisco_type_7',
+    message_digest_key_id:              '82',
+    message_digest_password:            '046E1803362E595C260E0B240619050A2D',
+    retransmit_interval:                '1000',
+    transmit_delay:                     '400',
+  },
+}
+
+#################################################################
+# TEST CASE EXECUTION
+#################################################################
+test_name "TestCase :: #{tests[:resource_name]}" do
+  # -------------------------------------------------------------------
+  device = platform
+  logger.info("#### This device is of type: #{device} #####")
+  logger.info("\n#{'-' * 60}\nSection 1. Default Property Testing")
+  test_harness_run(tests, :default)
+
+  id = :default
+  tests[id][:ensure] = :absent
+  tests[id].delete(:preclean)
+  test_harness_run(tests, id)
+
+  # -------------------------------------------------------------------
+  logger.info("\n#{'-' * 60}\nSection 2. Non Default Property Testing")
```

Comment:
We need title pattern testing for this provider.  Quite a few possibilities here.